docs/plugins/haveibeenpwned/plugin.js

```diff
@@ -0,0 +1,250 @@
+        statuses: [200, 404],
+        success: (data, xhr) => {
+            this.logger.info('xhr ' + JSON.stringify(xhr));
+            if (data && data.length > 0) {
+                this.logger.info('found breaches ' + JSON.stringify(data));
+                var breaches = "";
+                data.forEach(breach => { breaches += `<li>${breach.Name}</li>\n`; });
+                this._alert(`WARNING! This account has been pawned in the following breaches<br/>\n<ul>\n${breaches}\n</ul>\n<p>Please check on <a href="https://haveibeenpwned.com">https://haveibeenpwned.com</a>\n`);
+                this.userEditView.$el.focus();
+                this.userEditView.$el.addClass('input--error');
+                InputFx.shake(this.userEditView.$el);
+            } else {
+                this.logger.info("check pwnd name passed...");
+                this.userEditView.$el.removeClass('input--error');
+            }
+        },
+        error: (e, xhr) => {
+            let err = xhr.response && xhr.response.error || new Error('Network error');
+            this.logger.error('Pwned Password API error', 'GET', xhr.status, err);
+            err.status = xhr.status;
+        }
+    });   
+};
+
+DetailsView.prototype.checkPwdPwned = function (passwordHash) {
+    this.logger.info('check hibp pwd (hash) ' + passwordHash);
+    
+    prefix = passwordHash.substring(0, 5);
+    new HIBPUtils().xhrcall({
+        url: `https://api.pwnedpasswords.com/range/${prefix}`,
+        method: 'GET',
+        responseType: 'text',
+        headers: undefined,
+        data: null,
+        statuses: [200, 404],
+        success: data => {
+            if (data) {
+                this.logger.info('found breaches ' + JSON.stringify(data));
+                data.split('\r\n').forEach(line => {
+                    h = line.split(':');
+                    suffix = h[0]; nb = h[1];
+                    if (prefix + suffix === passwordHash) {
+                        this._alert(`WARNING: This password is referenced as pawned ${nb} times on <a href="https://haveibeenpwned.com">https://haveibeenpwned.com</a>!\n`);
+                        this.passEditView.$el.focus();
+                        this.passEditView.$el.addClass('input--error');
+                        InputFx.shake(this.passEditView.$el);
+                    }
+                });
+            } else {
+                this.logger.info("check pwnd passwd passed...");
+                this.passEditView.$el.removeClass('input--error');
+            }    
+        },
+        error: (e, xhr) => {
+            let err = xhr.response && xhr.response.error || new Error('Network error');
+            this.logger.error('Pwned Password API error', 'GET', xhr.status, err);
+            err.status = xhr.status;
+        }
+    });
+};
+
+DetailsView.prototype.fieldChanged = function (e) {
+    //this.logger.info('field changed ' + new HIBPUtils().stringify(e));
+    detailsViewFieldChanged.apply(this, [e]);
```

Comment:
Please replace with `detailsViewFieldChanged.apply(this, arguments);`, in case of future changes of this method.

Reply:
done